### ISSUE TITLE
module packaging tool "rollup"  and code compression tool "terser"

### ISSUE THREAD

**Issue by masx200** [x3dom/x3dom#1001]
It is recommended to use the module packaging tool "rollup"
It is recommended to use the code compression tool "terser"

**Comment by tsturm** [comment]
Both tools are build to work with modern javascript frameworks.
The whole x3dom codebase is over ten years old and currently we don't use any es2015+ features.
Until we refactor the whole codebase there is no need to use rollup or terser.

**Comment by masx200** [comment]
> Both tools are build to work with modern javascript frameworks.
> The whole x3dom codebase is over ten years old and currently we don't use any es2015+ features.
> Until we refactor the whole codebase there is no need to use rollup or terser.

You can use 'es2015+' in case of 'core-js', 'babel-polyfill', '@babel/core'.

And you should wrap closures outside the code to prevent global variable pollution.

**Comment by tsturm** [comment]
Yes, I know that. I have multiple modern javascript projects.
But in case of x3dom it's a huge effort to get the whole codebase to that level.

**Comment by tsturm** [comment]
A first version of a completely Node-based Build-System using ```terser``` is now available in the [new_build_system ](https://github.com/x3dom/x3dom/tree/new_build_system) branch.

```git checkout origin/new_build_system```

```npm install```

```npm run build```

**Comment by jamesleesaunders** [comment]
@tsturm Nice work! It is great to see x3dom build process move to more modern ways :-)

**Comment by jamesleesaunders** [comment]
If any help, I have had a little play with the new build process and had a quick test of the resulting dist files and all seems to work great, I have not run though all the tests but a few spot checks.

**Comment by tsturm** [comment]
@jamesleesaunders Thanks for the testing!
I have checked the tests/examples inside the repository too and it looks fine.

The dailybuild of the x3dom-server is already using the new build-system.

Before I can merge the branch into master I have to sync with @andreasplesch because he must update his build system, too.

**Comment by andreasplesch** [comment]
Please go ahead and merge into the master branch. I have disabled the CI on travis and netlify for now, for the master branch.

It may be overkill to use both travis/surge and netlify in parallel for automated building/deploy. Netlify is more convenient, so I will update that first when I have some time. It should be pretty straightforward. At this point I lean towards removing travis/surge.

**Comment by tsturm** [comment]
Okay, I have merged the new build system to master now.

**Comment by andreasplesch** [comment]
I have updated the netlify CI setup to use the new build system: https://app.netlify.com/sites/x3dom/overview
https://x3dom.netlify.com/
https://github.com/andreasplesch/x3dom/blob/master/netlify.toml
It seems to works well.
